src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/MonadBlueprint.hs

```diff
@@ -21,25 +21,23 @@ import           Data.Set                                            (Set)
 import qualified Data.Set                                            as Set
 import           Numeric.Natural                                     (Natural)
 import           Prelude                                             hiding (Bool (..), Eq (..), replicate, (*), (+), (-))
-import qualified Prelude                                             as Haskell
 
 import           ZkFold.Base.Algebra.Basic.Class
-import           ZkFold.Base.Algebra.Basic.Scale                     (Self (..))
 import           ZkFold.Base.Algebra.Polynomials.Multivariate        (var)
 import           ZkFold.Prelude                                      (replicate)
 import qualified ZkFold.Symbolic.Compiler.ArithmeticCircuit.Internal as I
 import           ZkFold.Symbolic.Compiler.ArithmeticCircuit.Internal hiding (constraint)
-import           ZkFold.Symbolic.Data.Bool                           (Bool (..), BoolType (..))
+import           ZkFold.Symbolic.Data.Bool                           (Bool (..))
 import           ZkFold.Symbolic.Data.Conditional                    (Conditional (..))
 import           ZkFold.Symbolic.Data.Eq                             (Eq (..))
 
-type WitnessField x a = (Algebra x a, FiniteField x, BinaryExpansion x,
+type WitnessField a x = (Algebra a x, FiniteField x, BinaryExpansion x,
```

Comment:
Why was this change made?

Reply:
The idea is that the last type parameter should be the "main" one (in this case -- the carrier of an algebraic structure)
But I guess there is no problem in switching places

Reply:
I think this one is fine. I agree with @TurtlePU's logic.

I was thinking about `Exponent b a`, though. The logic was that the order of type arguments is the same as in the mathematical notation of the related expressions (`Scale b a`, but `Exponent a b`). What do you think? Is it better after the change of before?

Reply:
Oh, actually `Exponent a b` makes more sense
When I myself was preparing this PR, I would mess the order up sometimes, so I guess `Exponent a b` is more intuitive, will change to that

Reply:
@vlasin, changed the order to `Exponent a b`